crates/transaction-emitter/src/lib.rs

```diff
@@ -270,6 +234,69 @@ impl SubmissionWorker {
         self.accounts
     }
 
+    async fn update_stats(
+        &mut self,
+        start_time: Instant,
+        txn_offset_time: u64,
+        num_requests: usize,
+    ) {
+        match wait_for_accounts_sequence(
+            &self.client,
+            &mut self.accounts,
+            self.params.txn_expiration_time_secs,
+        )
+        .await
+        {
+            Ok(()) => {
+                let latency = (Instant::now() - start_time).as_millis() as u64
+                    - txn_offset_time / num_requests as u64;
+                self.stats
+                    .committed
+                    .fetch_add(num_requests as u64, Ordering::Relaxed);
+                self.stats
+                    .latency
+                    .fetch_add(latency * num_requests as u64, Ordering::Relaxed);
+                self.stats
+                    .latencies
+                    .record_data_point(latency, num_requests as u64);
+            }
+            Err(committed_txn_info) => {
+                let num_committed = committed_txn_info
```

Comment:
This doesn't look right to me. We should only add the committed txns incrementally right?

Reply:
You're right, the original code was legit, I needed to look closer at gen_requests. I'll fix that up now.